### PR TITLE
SC-5915: homework dashboard

### PR DESCRIPTION
**Pull request by haniapalac**
# Description

## Links to Tickets or other pull requests

Dropdown fix will be done in ticket https://ticketsystem.schul-cloud.org/browse/SC-5858
## Changes

image

1. the title of the card was duplicating the Aufgabe offnen link,, changed to a div
2. Visual focus added for the Aufgabe offnen link 
3. Contrast fixed for the date section

image
4. Fixed opening of the sorting chip
5. Improved the add homework button
6. Improved the order of tabbing for the sorting, filtering and add button section
7. Added programmaticly adding of the aria-label for the closing button of the sorting chip

image
8. Improved pagination contrast

9. Improved placing of the sidebar skip link

## Screenshots of UI changes

See above

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)
https://docs.schul-cloud.org/pages/viewpage.action?spaceKey=CAP&title=Acceptance+Criteria

### CODE REVIEW

**Comment by haniapalac** [comment]
image
There is an icon missing on the button. What should I do with it? Do we want to replace it?

File: locales/de.json [hunk 1]
```diff
@@ -2290,7 +2290,10 @@
 			}
 		},
 		"footer": {
-			"ariaLabel": "footer",
+			"ariaLabel": {
```
Comment (ABialoskorski) [inline locales/de.json:2293]: please use aria_label, according to 
https://docs.schul-cloud.org/pages/viewpage.action?pageId=113643606
image

File: views/homework/components/overview_card_template.hbs [hunk 2]
```diff
@@ -67,55 +67,55 @@
                 {{{ truncateHTML (conflictFreeHtml this.description) 140 false }}}
             {{/if}}
         </div>
-    </a>
+    </div>
     <div>
         {{#inArray this.currentUser._id this.archived}}
-        <span class="count-badge" data-badge="Archiviert" /></span>
+        <span class="count-badge" data-badge="Archiviert" aria-label="Die Aufgabe ist archiviert" role="img" /></span>
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:73]: please add i18 here :)

File: views/homework/components/overview_card_template.hbs [hunk 3]
```diff
@@ -67,55 +67,55 @@
...
         {{else}}
             {{#if ../this.private}}
-            <span class="count-badge" data-badge="Entwürfe" /></span>
+            <span class="count-badge" data-badge="Entwürfe" aria-label="Die Aufgabe entwürfe" role="img" /></span>
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:76]: here as well as: "Die Aufgabe ist ein Entwurf"

File: views/homework/components/overview_card_template.hbs [hunk 4]
```diff
@@ -67,55 +67,55 @@
...
             {{else}}
-            <span class="count-badge" data-badge="Gestellt" /></span>
+            <span class="count-badge" data-badge="Gestellt" aria-label="Die Aufgabe ist gestellt" role="img" /></span>
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:78]: i18 please :)

File: views/homework/components/overview_card_template.hbs [hunk 5]
```diff
@@ -67,55 +67,55 @@
...
             {{/if}}
         {{/inArray}}
         {{#if this.grade}}
-            <span class="count-badge" data-badge="Feedback" /></span>
+            <span class="count-badge" data-badge="Feedback" aria-label="Die Aufgabe hat Feedback" role="img" /></span>
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:82]: see above

File: views/homework/components/overview_card_template.hbs [hunk 6]
```diff
@@ -67,55 +67,55 @@
...
         {{/if}}
     </div>
     <div class="action-group">
         {{#userHasPermission "HOMEWORK_CREATE"}}
             {{#inArray ../this.currentUser._id ../this.archived}}
                 <a class="btn btn-secondary btn-sm btn-archive" href="{{../url}}" target="_blank" data="archive=open"
-                    title="Verschiebe Aufgabe zwischen Übersichtsseite und Archiv" alt-text="<i class='fa fa-archive'></i>">
-                    <i class='fa fa-mail-reply'></i>
+                        aria-label="Verschiebe Aufgabe zwischen Übersichtsseite und Archiv {{$t "global.openInNewTabWarning"}}" alt-text="<i class='fa fa-archive' aria-hidden="true"></i>">
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:89]: add i18 please

File: views/homework/components/overview_card_template.hbs [hunk 7]
```diff
@@ -67,55 +67,55 @@
...
+                    <i class='fa fa-mail-reply' aria-hidden="true"></i>
                 </a>
             {{/inArray}}
             {{! Maybe Teachers create Tasks for other Teachers -> prevent editing by them }}
             {{#if ../this.isTeacher}}
                 {{#notInArray ../this.currentUser._id ../this.archived}}
                     <a class="btn btn-secondary btn-sm btn-archive" href="{{../url}}" target="_blank" data="archive=done"
-                        title="Verschiebe Aufgabe zwischen Übersichtsseite und Archiv" alt-text="<i class='fa fa-mail-reply'></i>">
-                        <i class='fa fa-archive'></i>
+                            aria-label="Verschiebe Aufgabe zwischen Übersichtsseite und Archiv {{$t "global.openInNewTabWarning"}}" alt-text="<i class='fa fa-mail-reply'></i>">
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:97]: for here as well :)

File: views/homework/components/overview_card_template.hbs [hunk 8]
```diff
@@ -67,55 +67,55 @@
...
+                        <i class='fa fa-archive' aria-hidden="true"></i>
                     </a>
                 {{/notInArray}}
                 {{#each ../this.actions}}
                     {{#if this.method}}
                         <a class="btn btn-secondary btn-sm {{this.class}}" href="{{this.link}}"
                             redirect="?p={{@root.pagination.currentPage}}" target="_blank" data-method="{{this.method}}"
-                            data-name="{{../../name}}" title="Die Aufgabe: '{{{escapeHtml ../../name}}}' {{{escapeHtml this.title}}}.">
-                            <i class="fa fa-{{this.icon}}"></i>
+                            data-name="{{../../name}}" aria-label="Die Aufgabe: '{{{escapeHtml ../../name}}}' {{{escapeHtml this.title}}}. {{$t "global.openInNewTabWarning"}}">
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:105]: same as above

File: views/homework/components/overview_card_template.hbs [hunk 9]
```diff
@@ -67,55 +67,55 @@
...
+                            <i class="fa fa-{{this.icon}}" aria-hidden="true"></i>
                         </a>
                     {{else}}
                         <a class="btn btn-secondary btn-sm {{this.class}}" href="{{this.link}}"
-                            title="Die Aufgabe: '{{{escapeHtml ../../name}}}' {{this.title}}">
-                            <i class="fa fa-{{this.icon}}"></i> {{this.label}}
+                            aria-label="Die Aufgabe: '{{{escapeHtml ../../name}}}' {{this.title}} {{$t "global.openInNewTabWarning"}}"">
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:110]: i18😁

File: views/homework/components/overview_card_template.hbs [hunk 10]
```diff
@@ -67,55 +67,55 @@
...
+                            <i class="fa fa-{{this.icon}}" aria-hidden="true"></i> {{this.label}}
                         </a>
                     {{/if}}
                 {{/each}}
             {{/if}}
         {{/userHasPermission}}
     </div>
-    <a href="{{this.url}}" title="Details der Aufgabe: '{{{escapeHtml this.name}}}'">
+    <a href="{{this.url}}" aria-label="Details der Aufgabe: '{{{escapeHtml this.name}}}'">
```
Comment (luizzaly) [inline views/homework/components/overview_card_template.hbs:118]: see above ☝🏼

File: views/lib/components/pagination.hbs [hunk 11]
```diff
@@ -3,39 +3,39 @@
         <ul class="pagination">
             {{#each pageItems}}
                 {{#if first}}
-                    <li class="page-item{{#if disabled}} disabled{{/if}}">
+                    <li class="page-item{{#if disabled}} disabled {{/if}}">
                         <a class="page-link" href="{{url}}">
-                            <span>&laquo;</span>
+                            <span class="{{#if disabled}} text-muted{{/if}}" role="img" aria-label="first page">&laquo;</span>
```
Comment (luizzaly) [inline views/lib/components/pagination.hbs:8]: please keep the aria-label in german: "Erste Seite" and add it to i18

File: views/lib/components/pagination.hbs [hunk 12]
```diff
@@ -3,39 +3,39 @@
...
                         </a>
                     </li>
                 {{/if}}
                 {{#if previous}}
                     <li class="page-item{{#if disabled}} disabled{{/if}}">
                         <a class="page-link" href="{{url}}">
-                            <span>&lsaquo;</span>
+                            <span class="{{#if disabled}} text-muted{{/if}}" role="img" aria-label="previous page">&lsaquo;</span>
```
Comment (luizzaly) [inline views/lib/components/pagination.hbs:15]: as above☝🏼 "Vorherige Seite"

File: views/lib/components/pagination.hbs [hunk 13]
```diff
@@ -3,39 +3,39 @@
...
+                    <li class="page-item{{#if disabled}} disabled text-muted{{/if}}{{#if active}} active{{/if}}">
                         <a class="page-link" href="{{url}}">{{num}}</a>
                     </li>
                 {{/if}}
                 {{#if next}}
-                    <li class="page-item{{#if disabled}} disabled{{/if}}">
+                    <li class="page-item{{#if disabled}} disabled {{/if}}">
                         <a class="page-link" href="{{url}}">
-                            <span>&rsaquo;</span>
+                            <span class="{{#if disabled}} text-muted{{/if}}" role="img" aria-label="next page">&rsaquo;</span>
```
Comment (luizzaly) [inline views/lib/components/pagination.hbs:27]: & here: "letzte Seite"

File: theme/thr/views/lib/footer.hbs [hunk 14]
```diff
@@ -32,9 +32,9 @@
 				</ul>
 			</nav>
 			<div>
-				<p lang="en" class="made-with-love">Made with <span class="heart" aria-label="love">❤</span> in Potsdam | Translated by <a
+				<p lang="en" class="made-with-love">Made with <span class="heart" aria-label={{$t "lib.footer.ariaLabel.footer"}} role="img">❤</span> in Potsdam | Translated by <a
```
Comment (luizzaly) [inline theme/thr/views/lib/footer.hbs:35]: please change to `{{$t "lib.footer.ariaLabel.love" }}`

**Comment by luizzaly** [comment]
> image
> There is an icon missing on the button. What should I do with it? Do we want to replace it?

Did you try to use a <span> or a <button> instead of the <a> maybe that will help to make the icon visible again...

**Review by pawelchoniawko — APPROVED** [review]

**Review by monaweitzenberg — APPROVED** [review]

**Pull request merged**